sys/net/link_layer/l2filter/l2filter.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_l2filter
+ * @{
+ *
+ * @file
+ * @brief       Link layer address filter implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "net/l2filter.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static inline bool match(const l2filter_t *filter,
+                         const void *addr, size_t addr_len)
+{
+    return ((filter->addr_len == addr_len) &&
+            (memcmp(filter->addr, addr, addr_len) == 0));
+}
+
+void l2filter_init(l2filter_t *list)
+{
+    assert(list);
+
+    for (unsigned i = 0; i < L2FILTER_LISTSIZE; i++) {
+        list[i].addr_len = 0;
+    }
+}
+
+int l2filter_add(l2filter_t *list, const void *addr, size_t addr_len)
```

Comment:
This function does not check for duplicate entries. Comparing the IPv6 blacklist module they used a bitfield to detect existing entries, for performance reasons. I can add check later, if you don't have time. @haukepetersen, @miri64 what's you opinion. Should we rather
1. not check for duplicates
2.  iterate through the list (via '`match()` function)
3. add a bitfield as done in IPv6 blacklist
IMO we should give 2. a try.